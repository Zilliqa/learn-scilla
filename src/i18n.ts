import i18next from 'i18next';

import en from './locales/en.json';
import es from './locales/es.json';
import ja from './locales/ja.json';
import ko from './locales/ko.json';
import zh from './locales/zh.json';

export const langDictionary = {
  en: 'English [en]',
  es: 'Español [es]',
  ja: '日本語 [ja]',
  ko: '한국어 [ko]',
  zh: '中文 [zh]'
};

const isDev: boolean = process.env.NODE_ENV === 'development';

i18next.init({
  lng: 'en',
  debug: isDev,
  react: {
    wait: false,
    bindI18n: 'languageChanged loaded',
    bindStore: 'added removed',
    nsMode: 'default'
  },
  interpolation: { escapeValue: false }, // not needed for react
  // lowerCaseLng: true,
  // have a common namespace used around the full app
  ns: ['translations'],
  defaultNS: 'translations',

  resources: {
    en: { translations: en },
    es: { translations: es },
    ja: { translations: ja },
    ko: { translations: ko },
    zh: { translations: zh }
  }
});

export default i18next;
