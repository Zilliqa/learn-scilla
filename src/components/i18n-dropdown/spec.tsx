import React from 'react';
import I18nDropdown from '.';
import { langDictionary } from '../../i18n';
import * as renderer from 'react-test-renderer';
import i18n from '../../i18n';

it('renders correctly', () => {
  const tree = renderer
    .create(<I18nDropdown i18n={i18n} langDictionary={langDictionary} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
