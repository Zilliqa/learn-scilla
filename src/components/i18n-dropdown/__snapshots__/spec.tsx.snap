// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n Dropdown tests basic tests matches the snapshot 1`] = `
<li
  className="dropdown nav-item"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle nav-link"
    href="#"
    onClick={[Function]}
  >
    English [en]
  </a>
  <div
    aria-hidden={true}
    className="dropdown-menu dropdown-menu-right"
    role="menu"
    size="sm"
    tabIndex="-1"
  >
    <button
      aria-label="English [en]"
      className="btn btn-md btn-transparent btn-block text-left"
      onClick={[Function]}
    >
       
      English [en]
       
    </button>
    <button
      aria-label="中文 [zh]"
      className="btn btn-md btn-transparent btn-block text-left"
      onClick={[Function]}
    >
       
      中文 [zh]
       
    </button>
    <button
      aria-label="日本語 [ja]"
      className="btn btn-md btn-transparent btn-block text-left"
      onClick={[Function]}
    >
       
      日本語 [ja]
       
    </button>
  </div>
</li>
`;
